pysr/package_compiler.py

```diff
@@ -0,0 +1,115 @@
+    _get_io_arg,
+    _process_julia_project,
+    _import_error,
+)
+
+
+def _add_sr_to_julia_project(Main, io_arg):
+    Main.eval("using Pkg")
+    Main.sr_spec = Main.PackageSpec(
+        name="SymbolicRegression",
+        url="https://github.com/MilesCranmer/SymbolicRegression.jl",
+        rev="v" + __symbolic_regression_jl_version__,
+    )
+    Main.clustermanagers_spec = Main.PackageSpec(
+        name="ClusterManagers",
+        rev="v0.4.2",
+    )
+    Main.packagecompiler_spec = Main.PackageSpec(
+        name="PackageCompiler",
+        rev="v2.1.0",
+    )
+    Main.pycall_spec = Main.PackageSpec(
+        name="PyCall",
+        rev="v1.94.1",
+    )
+    Main.eval(
+        "Pkg.add(["
+        + ", ".join(
+            ["sr_spec", "clustermanagers_spec", "packagecompiler_spec", "pycall_spec"]
+        )
+        + f"], {io_arg})"
+    )
+    Main.eval(f'Pkg.build("PyCall", {io_arg})')
+
+
+def _update_julia_project(Main, is_shared, io_arg):
+    try:
+        if is_shared:
+            _add_sr_to_julia_project(Main, io_arg)
+        Main.eval("using Pkg")
+        Main.eval(f"Pkg.resolve({io_arg})")
+    except (JuliaError, RuntimeError) as e:
+        raise ImportError(_import_error()) from e
+
+
+def install(julia_project=None, quiet=False):  # pragma: no cover
+    """
+    Install PyCall.jl and all required dependencies for SymbolicRegression.jl.
+
+    Also updates the local Julia registry.
+    """
+    import julia
+
+    _julia_version_assertion()
+    # Set JULIA_PROJECT so that we install in the pysr environment
+    processed_julia_project, is_shared = _process_julia_project(julia_project)
+    _set_julia_project_env(processed_julia_project, is_shared)
+
+    julia.install(quiet=quiet)
+    Main = init_julia(julia_project, quiet=quiet, use_sysimage=False)
+    io_arg = _get_io_arg(quiet)
+
+    if is_shared:
+        # Install SymbolicRegression.jl:
+        _add_sr_to_julia_project(Main, io_arg)
+
+    Main.eval("using Pkg")
+    Main.eval(f"Pkg.instantiate({io_arg})")
+    Main.eval(f"Pkg.precompile({io_arg})")
+
+    if not quiet:
+        warnings.warn(
+            "It is recommended to restart Python after installing PySR's dependencies,"
+            " so that the Julia environment is properly initialized."
+        )
+
+
+def compile(
+    julia_project=None,
+    quiet=False,
+    sysimage_name="pysr.so",
+):
+    """Create a PackageCompiler.jl sysimage for SymbolicRegression.jl."""
+    Main = init_julia(
+        julia_project=julia_project,
+        quiet=quiet,
+        use_sysimage=False,
+        julia_kwargs={
+            "compiled_modules": False,
+            "optimize": 3,
+            "threads": cpu_count(),
+            "compile": "all",
+        },
+    )
+    cur_project_dir = Main.eval("dirname(Base.active_project())")
+    sysimage_path = str(Path(cur_project_dir) / sysimage_name)
+    from julia import PackageCompiler
+
+    Main.eval("using SymbolicRegression")
```

Comment:
Could add a call here:

```python
from . import PySRRegressor
PySRRegressor(verbosity=0, progress=False, temp_equation_file=True).fit([[1]], [1])
```

just so that the PyCall interface specific to PySR gets compiled too.